File: testdata/bench/c/BenchmarkTest00057.c
#include <stdio.h>
#include <string.h>

void safe_open(const char *filename) {
    if (strstr(filename, "..") != NULL) return;
    char path[512];
    snprintf(path, sizeof(path), "/data/%s", filename);
    FILE *fp = fopen(path, "r");
    if (fp) fclose(fp);
}

File: testdata/bench/swift/BenchmarkTest00057.swift
import Vapor
import Foundation

func handler057(_ req: Request) throws -> Response {
    let filename = try req.parameters.require("file")
    let safeDir = URL(fileURLWithPath: "/var/data/public/")
    let fileURL = safeDir.appendingPathComponent(filename).standardizedFileURL
    guard fileURL.path.hasPrefix(safeDir.path) else { throw Abort(.forbidden) }
    let data = try Data(contentsOf: fileURL)
    return Response(status: .ok, body: .init(data: data))
}
